packages/admin-next/dashboard/src/v2-routes/sales-channels/sales-channel-add-products/components/add-products-to-sales-channel-form.tsx

```diff
@@ -0,0 +1,358 @@
+import { zodResolver } from "@hookform/resolvers/zod"
+import { Product, SalesChannel } from "@medusajs/medusa"
+import { Button, Checkbox, Hint, Table, Tooltip, clx } from "@medusajs/ui"
+import {
+  PaginationState,
+  RowSelectionState,
+  createColumnHelper,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table"
+import {
+  adminProductKeys,
+  adminSalesChannelsKeys,
+  useAdminCustomPost,
+  useAdminProducts,
+} from "medusa-react"
+import { useEffect, useMemo, useState } from "react"
+import { useForm } from "react-hook-form"
+import { useTranslation } from "react-i18next"
+import * as zod from "zod"
+import {
+  ProductAvailabilityCell,
+  ProductCollectionCell,
+  ProductStatusCell,
+  ProductTitleCell,
+  ProductVariantCell,
+} from "../../../../components/common/product-table-cells"
+import { OrderBy } from "../../../../components/filtering/order-by"
+import { Query } from "../../../../components/filtering/query"
+import { LocalizedTablePagination } from "../../../../components/localization/localized-table-pagination"
+import {
+  RouteFocusModal,
+  useRouteModal,
+} from "../../../../components/route-modal"
+import { useQueryParams } from "../../../../hooks/use-query-params"
+import { queryClient } from "../../../../lib/medusa"
+
+type AddProductsToSalesChannelFormProps = {
+  salesChannel: SalesChannel
+}
+
+const AddProductsToSalesChannelSchema = zod.object({
+  product_ids: zod.array(zod.string()).min(1),
+})
+
+const PAGE_SIZE = 50
+
+export const AddProductsToSalesChannelForm = ({
+  salesChannel,
+}: AddProductsToSalesChannelFormProps) => {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+
+  const form = useForm<zod.infer<typeof AddProductsToSalesChannelSchema>>({
+    defaultValues: {
+      product_ids: [],
+    },
+    resolver: zodResolver(AddProductsToSalesChannelSchema),
+  })
+
+  const { setValue } = form
+
+  const { mutateAsync, isLoading: isMutating } = useAdminCustomPost(
+    `/admin/sales-channels/${salesChannel.id}/products/batch/add`,
+    [
+      adminSalesChannelsKeys.lists(),
+      adminSalesChannelsKeys.detail(salesChannel.id),
+    ]
+  )
+
+  const [{ pageIndex, pageSize }, setPagination] = useState<PaginationState>({
```

Comment:
Definitely not for now, but I think we should create a Table component that is a bit more high-level and takes care of pagination internally, so we can skip adding similar logic all over the place and instead wrap react-table for the most common use-cases.